### PR TITLE
Support admin endpoint for updating the logging level

### PR DESCRIPTION
**Pull request by LisaLudique**
This change adds support for an admin endpoint that updates the logging level, via internally instantiating a new logger at the specified level. Currently it does not restrict the possible zap logging levels the logger can be set to.

It's worth noting that updates to the logging level are not reflected in the `/server_info` endpoint, which only reflects the bootstrap config that the server is started with (unlike envoy's `/config_dump` which reflects the current state).

### CODE REVIEW

File: internal/app/admin/http/handler.go [hunk 1]
```diff
@@ -34,6 +39,11 @@ func getHandlers(bootstrap *bootstrapv1.Bootstrap, orchestrator *orchestrator.Or
 			"print cache entry for a given key. usage: `/cache/<key>`",
 			cacheDumpHandler(orchestrator),
 		},
+		{
+			"/log_level/",
+			"update the log level. usage: `/log_level/<level>`",
```
Comment (jessicayuen) [inline internal/app/admin/http/handler.go:44]: May want to consider listing the options here: debug, info, warn, error, fatal

File: internal/app/admin/http/handler.go [hunk 2]
```diff
@@ -126,11 +138,29 @@ func resourceToString(resource cache.Resource) (string, error) {
...
 		return splitPath[2], nil
 	}
-	return "", fmt.Errorf("unable to parse cache key from path: %s", path)
+	return "", fmt.Errorf("unable to parse parameter from path: %s", path)
+}
+
+func logLevelHandler(l log.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == "POST" {
+			logLevel, _ := getParam(req.URL.Path)
```
Comment (jessicayuen) [inline internal/app/admin/http/handler.go:153]: If this returns an error I think we should exit. `zap.ParseLogLevel` will default to Info IIRC.  May produce unexpected behavior.

File: internal/app/admin/http/handler.go (hunk 1 above)
Comment (LisaLudique) [reply internal/app/admin/http/handler.go:44]: Is it worth restricting the possible levels to be consistent with bootstrap.proto? What's the reasoning for restricting the possible levels?

File: internal/app/admin/http/handler.go (hunk 2 above)
Comment (LisaLudique) [reply internal/app/admin/http/handler.go:153]: I think that's what it currently does? Returns a BadRequest error to the client

Comment (jessicayuen) [reply internal/app/admin/http/handler.go:153]: we don't assign the error in this line though. If we can't parse the param in `getParam`, `zap.ParseLogLevel("")` is effectively called, which defaults to the info log level, no?

Comment (LisaLudique) [reply internal/app/admin/http/handler.go:153]: oh, I was looking at the wrong line. Yeah we should catch that error I don't know why I didn't include it.

Comment (jessicayuen) [reply internal/app/admin/http/handler.go:153]: actually nvm, I think this is me misreading the implementation for `zap.ParseLogLevel`.

File: internal/app/admin/http/handler.go (hunk 1 above)
Comment (jessicayuen) [reply internal/app/admin/http/handler.go:44]: I just meant listing the levels specified in the proto here to help guide the endpoint user easier (without having to search up the proto spec).

**Comment by jessicayuen** [comment]
lgtm, will +1 once you resolve the conflicts!

File: internal/app/admin/http/handler.go [hunk 3]
```diff
@@ -126,11 +138,33 @@ func resourceToString(resource cache.Resource) (string, error) {
...
+	return "", fmt.Errorf("unable to parse parameter from path: %s", path)
+}
+
+func logLevelHandler(l log.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == "POST" {
+			logLevel, err := getParam(req.URL.Path)
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				fmt.Fprintf(w, "unable to parse log level from path: %s", err.Error())
```
Comment (jessicayuen) [inline internal/app/admin/http/handler.go:156]: missing `return`?

**Pull request merged**